### DISCUSSION TITLE
Embedd nvc - libnvc as shared library

### DISCUSSION THREAD

**Discussion by lemmuh** [nickg/nvc#872] [Q&A] [answered]
How can I compile the libnvc as shared library in order to embbed the nvc functionality into my own main application ?

**Comment by nickg** [comment, marked as answer]
This isn't supported: you need to do it the other way round and compile your application to a shared library which you load as a plugin with `--load`.

The existing static library is just a convineince for the build system and not something that is exposed to end users. Even if there was a shared library with stable interfaces I doubt it would be useful to most users as the resultant program would be covered by the GPL and couldn't be distributed without the corresponding sources.
